Lesson 2/app/src/main/java/com/mobileapps/lesson2/SignUpActivity.kt

```diff
@@ -12,13 +11,16 @@ class SignUpActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_sign_up)
 
-        val loginTextView: TextView = findViewById(R.id.loginTextView)
-        loginTextView.setOnClickListener {
-            val intent = Intent(this, AccountActivity::class.java)
-            startActivity(intent)
+        if (savedInstanceState == null) {
+            replaceFragment(RegisterFragment())
         }
 
     }
+    private fun replaceFragment(fragment: Fragment) {
```

Comment:
Think about backstack management for organic navigation. If I press 'register' by mistake, I'd like to press back button to go back, not close the app